packages/app-dashboard/src/components/withdraw/WalletConnect/WalletConnectUtil.ts

```diff
@@ -0,0 +1,290 @@
+
+  // Get or create the client
+  const client = await createWalletConnectClient();
+
+  if (!client) {
+    throw new Error('WalletKit client not initialized');
+  }
+
+  if (sessionSigs) {
+    try {
+      if (pkpWallet) {
+        pkpWallet = null;
+      }
+
+      pkpWallet = await createPKPWallet(agentPKP, sessionSigs);
+    } catch (error) {
+      console.error('Failed to create PKP wallet:', error);
+      throw new Error('Failed to create PKP wallet');
+    }
+  }
+
+  return {
+    address: agentPKP.ethAddress,
+    publicKey: agentPKP.publicKey,
+  };
+}
+
+/**
+ * Get the initialized PKP wallet instance
+ * @returns The PKP wallet instance or null if not initialized
+ */
+export function getPKPWallet(): PKPEthersWallet | null {
+  return pkpWallet;
+}
+
+/**
+ * Initialize WalletKit directly without any React hooks
+ * @param setClient Optional callback to set the client in state
+ * @param forceReset Whether to force a reset of the client
+ * @returns The initialized WalletKit client
+ */
+export async function createWalletConnectClient(
+  setClient?: (client: IWalletKit) => void,
+  forceReset = false,
+): Promise<IWalletKit> {
+  // If we're forcing a reset, clean up the existing client
+  if (forceReset && walletKitClient) {
+    console.log('Force resetting WalletKit client');
+    await resetWalletConnectClient();
+  }
+
+  // If already initialized, return existing instance
+  if (walletKitClient) {
+    if (setClient) {
+      setClient(walletKitClient);
+    }
+    return walletKitClient;
+  }
+
+  // Prevent concurrent initialization
+  if (isInitializing) {
+    console.log('WalletKit initialization in progress, waiting...');
+    // Wait for initialization to complete
+    return new Promise<IWalletKit>((resolve) => {
+      const checkInterval = setInterval(() => {
+        if (walletKitClient && !isInitializing) {
+          clearInterval(checkInterval);
+          if (setClient) {
+            setClient(walletKitClient);
+          }
+          resolve(walletKitClient);
+        }
+      }, 100);
+    });
+  }
+
+  try {
+    isInitializing = true;
+    console.log('Initializing new WalletKit instance');
+
+    // Create a shared Core instance
+    const core = new Core({
+      projectId: VITE_WALLETCONNECT_PROJECT_ID,
+    });
+
+    // Initialize WalletKit with the required parameters - according to documentation
+    walletKitClient = await WalletKit.init({
+      core, // Pass the shared core instance
+      metadata: {
+        name: 'Vincent App',
+        description: 'Vincent App using PKP with WalletKit',
+        url: window.location.origin,
+        icons: ['https://lit.network/favicon.ico'], // TODO: Add a logo
+      },
+    });
+
+    // Store the client in the store if a setter was provided
+    if (setClient) {
+      setClient(walletKitClient);
+    }
+
+    return walletKitClient;
+  } catch (error) {
+    console.error('Error initializing WalletKit:', error);
+    throw error;
+  } finally {
```

Comment:
Don't just log and re-throw. If the error is not handled just let it bubble up without catching or the logs will pollute with a bunch of repeated logs real quick